src/shims/env.rs

```diff
@@ -348,16 +354,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
         this.assert_target_os("windows", "GetCurrentDirectoryW");
 
-        this.check_no_isolation("`GetCurrentDirectoryW`")?;
-
-        let size = u64::from(this.read_scalar(size_op)?.to_u32()?);
-        let buf = this.read_scalar(buf_op)?.check_init()?;
-
-        // If we cannot get the current directory, we return 0
-        match env::current_dir() {
-            Ok(cwd) =>
-                return Ok(windows_check_buffer_size(this.write_path_to_wide_str(&cwd, buf, size)?)),
-            Err(e) => this.set_last_error_from_io_error(e)?,
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("GetCurrentDirectoryW", reject_with)?;
+            let err = Error::new(ErrorKind::NotFound, "rejected due to isolation");
+            this.set_last_error_from_io_error(err)?;
```

Comment:
Please return early in this branch, so that the code below does not have to be moved to the right.